crates/bevy_ui/src/flex/mod.rs

```diff
@@ -320,6 +328,11 @@ pub fn flex_node_system(
     // clean up removed nodes
     flex_surface.remove_entities(removed_nodes.iter());
 
+    // When a `CalculatedSize` component is removed from an entity, we need to remove the measure from the corresponding taffy node.
+    for entity in removed_calculated_sizes.iter() {
```

Comment:
What would you think about moving this into a separate system?

This one seems to be getting quite big and doing a lot of different things.

Reply:
Definitely needs to be done, and I found this bug while working on some other changes that require `flex_node_system` to be split up. I'm not quite sure how they'll work out yet though, so I just put in this PR that fixes the bug with minimal changes etc.